### PR TITLE
Fix shellcheck in `test/cmd/apply.sh`

### PR DESCRIPTION
**Pull request by xychu**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

/kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Part of https://github.com/kubernetes/kubernetes/issues/72956

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by xychu** [comment]
@BenTheElder plz take a look when it's convenience, if the changes are on the right track, I'll update other files under `test/cmd`

File: test/cmd/apply.sh [hunk 1]
```diff
@@ -27,13 +27,13 @@ run_kubectl_apply_tests() {
   kube::log::status "Testing kubectl apply"
   ## kubectl apply should create the resource that doesn't exist yet
   # Pre-Condition: no POD exists
-  kube::test::get_object_assert pods "{{range.items}}{{$id_field}}:{{end}}" ''
+  kube::test::get_object_assert pods "{{range.items}}{{${id_field:?}}}:{{end}}" ''
```
Comment (BenTheElder) [inline test/cmd/apply.sh]: why :? instead of just the braces fix?

Comment (xychu) [reply test/cmd/apply.sh]: Got https://github.com/koalaman/shellcheck/wiki/SC2154.
And from the wiki, it says:

> Note: This message only triggers for variables with lowercase characters in their name (foo and kFOO but not FOO) due to the standard convention of using lowercase variable names for unexported, local variables.

And:

> If you know for a fact that the variable is set, you can use `${var:?}` to fail if the variable is unset (or empty), or explicitly initialize/declare it with `var=""` or `declare var`. You can also disable the message with a directive.

I've tried changing variable name to uppercase and the `${var:?}`  way.
Not sure which is more suitable.

Comment (BenTheElder) [reply test/cmd/apply.sh]: this change is reasonable I suppose, but we are not doing this elsewhere 🤔

Comment (xychu) [reply test/cmd/apply.sh]: @BenTheElder FYI: https://github.com/kubernetes/kubernetes/pull/73518 is the uppercase way.

File: test/cmd/apply.sh [hunk 2]
```diff
@@ -46,18 +46,18 @@ run_kubectl_apply_tests() {
...
   kube::test::get_object_assert deployments "{{range.items}}{{$id_field}}{{end}}" 'test-deployment-retainkeys'
   # Post-Condition: deployment "test-deployment-retainkeys" has defaulted fields
-  [[ "$(kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]}" | grep RollingUpdate)" ]]
-  [[ "$(kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]}" | grep maxSurge)" ]]
-  [[ "$(kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]}" | grep maxUnavailable)" ]]
-  [[ "$(kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]}" | grep emptyDir)" ]]
+  kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]}" | grep -q RollingUpdate
+  kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]}" | grep -q maxSurge
+  kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]}" | grep -q maxUnavailable
+  kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]}" | grep -q emptyDir
```
Comment (php-coder) [inline test/cmd/apply.sh]: BTW is it intentionally that we execute the same command 4 times? It seems that we can execute it only once and later use its output for our checks. Something like this:
```bash
local deployment_yaml="$(kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]}" || true)"
echo "$deployment_yaml" | grep -q RollingUpdate
echo "$deployment_yaml" | grep -q maxSurge
echo "$deployment_yaml" | grep -q maxUnavailable
echo "$deployment_yaml" | grep -q emptyDir
```

Comment (xychu) [reply test/cmd/apply.sh]: Good point, will update in other pr since this will be closed as a dup of https://github.com/kubernetes/kubernetes/pull/73518

**Pull request closed without merging**